bindings/matrix-sdk-ffi/src/client.rs

```diff
@@ -108,10 +108,12 @@ impl Client {
             device_id,
             homeserver_url: _,
             is_soft_logout,
+            sliding_sync_proxy,
         } = session;
 
         // update soft logout state
         self.state.write().unwrap().is_soft_logout = is_soft_logout;
+        self.state.write().unwrap().sliding_sync_proxy = sliding_sync_proxy;
```

Comment:
General question - could I have done `let state = state.write().unwrap()` and then written both of these to that local var, or would that make a copy of the state meaning I wouldn't actually be writing to the one in the client?

Reply:
The local var would be a `RwLockWriteGuard` that has a `Deref` and `DerefMut` implementation for whatever `T` is in the `RwLock<T>` definition.

Generally in Rust, unless your type is `Copy`, no implicit copying is being done, you'd have to use `foo.clone()` or `foo.to_owned()` to create a copy.

To answer your question, yes this works just fine:

```rust
let lock = self.state.write().unwrap();

lock.is_soft_logout = is_soft_logout;
lock.sliding_sync_proxy = sliding_sync_proxy;
// Not sure if you need or want the explicit drop, depends on the size of your scope.
drop(lock)
```

Reply:
Super thanks 👍